src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -245,10 +248,23 @@ protected AIState decide()
      */
     protected EntityLivingBase getTarget()
     {
-        final AbstractBuildingGuards building = (AbstractBuildingGuards) getOwnBuilding();
+        final AbstractBuildingGuards building = getOwnBuilding();
 
-        if (building != null && target == null)
+        if (building != null && target == null && worker.getCitizenColonyHandler().getColony() != null)
         {
+            for (final CitizenData citizen : worker.getCitizenColonyHandler().getColony().getCitizenManager().getCitizens())
+            {
+                if (citizen.getCitizenEntity().isPresent())
+                {
+                    final EntityLivingBase entity = citizen.getCitizenEntity().get().getRevengeTarget();
+                    if (entity instanceof AbstractEntityBarbarian
+                          && worker.canEntityBeSeen(entity))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)